web-api/src/storage/postgres.rs

```diff
@@ -398,18 +391,13 @@ impl Database {
         Ok(())
     }
 
-    async fn upsert_tag_weights<'c, I>(
+    async fn upsert_tag_weights(
         tx: &mut Transaction<'_, Postgres>,
         user_id: &UserId,
-        updates: I,
-    ) -> Result<(), Error>
-    where
-        I: IntoIterator<Item = (&'c str, i32)>,
-    {
-        let updates = updates.into_iter();
-
+        updates: &HashMap<&str, i32>,
+    ) -> Result<(), Error> {
```

Comment:
As this was created from a map before this couldn't trigger the bug before, I still changed it to avoid running into the bug again in the future.